node/src/actors/chain_manager/mod.rs

```diff
@@ -1932,7 +1932,9 @@ fn update_pools(
     }
 
     for vt_tx in &block.txns.value_transfer_txns {
-        transactions_pool.vt_remove(&vt_tx.hash());
+        if transactions_pool.vt_contains(&vt_tx.hash()) {
+            transactions_pool.vt_remove(&vt_tx.hash());
```

Comment:
vt_removes also tries to remove the transactions that could have the same inputs that the consolidated transaction, so it would have to be applied without the contains, because it could happen that you can not have the transaction in your mempool but you would have another transactions with same inputs...

But, to enforce that, we should include the transaction in vt_remove instead of the hash. I could do that in other PR

Reply:
So, currently, in master, this transactions could stay in the mempool forever? Open an issue

Reply:
merged a PR :P

Reply:
I'll have a look at this and merge the newest master with my PR.

Reply:
Basically we realized that we must call `vt_remove` even if the transactions are not in the transactions pool, and that was a potential issue. You simply need to revert this line, and call `vt_remove` without checking `vt_contains`.

Reply:
Yes, I understood that from the discussion. 😉

Only one question, will vt_remove_inner no crash when the initial tx_key was not present? In other words will `self.vt_transactions.remove(key)` graciously stop or crash and burn?